views/js/runner/testStore.js

```diff
@@ -0,0 +1,295 @@
+     * @throws {TypeError} with no testId
+     */
+    return function testStoreLoader(testId, preselectedBackend){
+
+        var storeNames = [];
+        var volatiles = [];
+        var unifiedStore;
+
+        var getMode = Promise.resolve();
+        if(_.isUndefined(unifiedStore)){
+            getMode = selectStoreMode(testId, preselectedBackend).then(function(result){
+                unifiedStore = !!result;
+
+                logger.debug('Test store mode ' + (unifiedStore ? 'unified' : 'multiple (legacy)') + ' for ' + testId);
+            });
+        }
+
+        if(_.isEmpty(testId)){
+            throw new TypeError('The store must be identified with a unique test identifier');
+        }
+
+        /**
+         * Wraps a store and add the support of "volatile" storages
+         * @typedef {Object} testStore
+         */
+        return {
+
+            /**
+             * Get a wrapped store instance, that let's you use multiple stores inside one store...
+             * (or in multiple stores if the test is in legacy mode)
+             * @param {String} storeName - the name of the sub store
+             * @returns {Promise<storage>}
+             */
+            getStore : function getStore(storeName) {
+                if(_.isEmpty(storeName)){
+                    throw new TypeError('A store name must be provided to get the store');
+                }
+
+                if(!_.contains(storeNames, storeName)){
+                    storeNames.push(storeName);
+                }
+                return getMode.then(function(){
+                    if (unifiedStore) {
+                        return store(testId, preselectedBackend);
+                    } else {
+
+                        return store(storeName + '-' + testId, preselectedBackend);
+                    }
+                })
+                .then(function(loadedStore){
+                    var keyPattern = new RegExp('^' + storeName + '__');
+                    var storeKey = function storeKey(key){
+                        return unifiedStore ? storeName + '__' + key : key;
+                    };
+
+                    /**
+                     * The wrapped storage
+                     * @type {Object}
+                     */
+                    return {
+
+
+                        /**
+                         * Get an item with the given key
+                         * @param {String} key
+                         * @returns {Promise<*>} with the result in resolve, undefined if nothing
+                         */
+                        getItem : function getItem(key){
+                            return loadedStore.getItem(storeKey(key));
+                        },
+
+                        /**
+                         * Get all store items
+                         * @returns {Promise<Object>} with a collection of items
+                         */
+                        getItems : function getItems(){
```

Comment:
in the same fashion as above, I prefer avoiding having two methods with really close names. `getAllItems()` would mark the difference, for ex, but the obvious advantage here is that it matches the store api so maybe it's better to keep it this way.

Reply:
The goal is to wrap the `store` API, so I'll keep as it is. And in any case, I'm not really agree since the plural is sufficient to mark the difference.